### ISSUE TITLE
HDMI 4K enhanced mode as option?

### ISSUE THREAD

**Issue by rubin110** [miracle2k/onkyo-eiscp#145]
TLDR can this library issue commands to switch between HDMI 4K Enhanced and Standard modes?

My Onkyo TX-NR787 has a secret menu to [switch to"HDMI 4K Enhanced" mode](https://www.xlr8yourmac.com/audio/Onkyo_RZ_Receivers_Tips_Updates.html), which enables 10-bit/12-bit support. In order to turn this feature on I have to go to the physical unit, while holding the BD/DVD input button also press the power button twice, then go back to the input I was on originally. This is turned on for the whole unit and not just for a single input. There is no way to do this from the remote.

![image](https://user-images.githubusercontent.com/222551/230801265-fa368bab-d256-496a-bbee-57d141767cbe.png)

![image](https://user-images.githubusercontent.com/222551/230801276-0920d324-8321-439d-9db1-9a9062d1c361.png)

My Chromecast with Google TV in HDR mode only outputs YCbCr4:2:2 12 bit at 60 Hz, while my PC in HDR mode can only spit out YCbCr4:2:0 8 bit at 60 Hz. When the receiver is in enhanced mode I can only view my Google TV however the input my PC is on just displays a blue screen. I can switch back to standard mode which makes my PC work, but then my Google TV switches to non-HDR with no option to switch it back to HDR.

It would be great if I could issue commands from this library to switch these modes and then just automate turning it on and off. I dug around this project but haven't found anything that would be related to this option. Could this be added?

Thank you.

**Comment by mitchcapper** [comment]
Just posted an updating protocol documentation file which includes how to do this.  You can do it with raw commands right now:
```
"HSF" - HDMI 4K/8K Signal Format Command	
"00"	sets HDMI 4K/8K Signal Format:4K Standard
"01"	sets HDMI 4K/8K Signal Format:4K Enhanced
"02"	sets HDMI 4K/8K Signal Format:8K Standard
"03"	sets HDMI 4K/8K Signal Format:8K Enhanced
"UP"	sets HDMI 4K/8K Signal Format Wrap-Around Up
"QSTN"	gets HDMI 4K/8K Signal Format Status
```

**Comment by rubin110** [comment]
Thank you so much!

The irony is that I got so frustrated with Onkyo making this such a dumb
thing, that it prompted me to finally start my build-a-new-gaming-pc
project and run a newer GPU that can spit out what's needed for
enhanced mode. So I no longer have a need to auto switch the modes when
switching inputs.

Thank you though!

**Comment by rubin110** [comment]
I want to verify this so I can close out this ticket. Is it possible to do so with just the plain old `onkyo` binary?

**Comment by mitchcapper** [comment]
not sure the lib has receiver.raw that would work.  In terms of officially command support wont be until the new file submitted is parsed to update the command db.

**Comment by rubin110** [comment]
Coming back to this ticket. I don't have a good means of testing out this fix. Not sure if you would like me to close out this ticket or not.
